org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java

```diff
@@ -2954,10 +2961,22 @@ public ZipFile getZipFile(IPath path, boolean checkInvalidArchiveCache) throws C
 		}
 		ZipCache zipCache;
 		ZipFile zipFile;
+		File localFile = getLocalFile(path);
 		if ((zipCache = this.zipFiles.get()) != null && (zipFile = zipCache.getCache(path)) != null) {
-			return zipFile;
+			if (zipCache.lastModified(path) == localFile.lastModified()) {
+				return zipFile;
+			} else {
+				try {
+					zipFile.close();
```

Comment:
I'm not sure this is correct, as per design the owner of the zip file is supposed to close it. Here the close() may close the file even if it is still used by a client. So a simply "touch" on a file, followed by the call here may break some code now.